beacon-chain/core/incentives/incentives.go

```diff
@@ -113,22 +127,35 @@ func ApplyCrosslinkRewardsAndPenalties(
 	totalActiveValidatorDeposit uint64,
 	totalBalance uint64,
 	voteBalance uint64,
-) error {
+) ([]*pb.ValidatorRecord, error) {
```

Comment:
I think over here it would be better served to use this function from `validator.go` : 
```
func CopyValidators(validatorSet []*pb.ValidatorRecord) []*pb.ValidatorRecord
```
The same applies to all other instances below where you are copying the validator set.